wai-extra/Network/Wai/Middleware/Gzip.hs

```diff
@@ -189,11 +311,23 @@ compressFile s hs file cache sendResponse = do
                             Z.feedDeflate deflate bs >>= goPopper
                             loop
                     goPopper $ Z.finishDeflate deflate
-            either onErr (const onSucc) (x :: Either SomeException ()) -- FIXME bad! don't catch all exceptions like that!
```

Comment:
Is the FIXME not relevant?

Reply:
The change fixes the FIXME, no?
It only catches what could be thrown in the `try` block and rethrows everything else.

Or do you mean the comment should remain?
What if I added a comment like:
`We generally don't want to catch all exceptions, but we catch `SomeException` here to check on two types of exceptions and then rethrow any other exception. Effectively only catching the ones we want to catch.`

Reply:
No it's fine as is. I didn't grok on first read.